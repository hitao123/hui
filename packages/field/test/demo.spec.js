import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Field from '../index.vue';
import FieldDemo from '../demo/index.vue';

describe('Field component', () => {
  it('should show Field', () => {
    const warpper = shallowMount(Field, {
      propsData: {
        label: '姓名',
        placeholder: '请输入姓名'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Field Demo', () => {
    expect(renderToString(FieldDemo)).toMatchSnapshot();
  });
});
