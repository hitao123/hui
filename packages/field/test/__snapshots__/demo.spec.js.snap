// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="请输入用户名" class="h-cell h-hairline h-field">
        <!---->
        <!---->
        <div class="h-cell__value h-cell__value--alone">
          <div class="h-field__body"><input type="input" placeholder="请输入用户名" value="" class="h-field__control">
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="请输入用户名" class="h-cell h-cell--required h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>用户名</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="input" placeholder="请输入用户名" value="" class="h-field__control">
            <!---->
            <div class="h-field__icon"><i class="h-icon h-icon-question" style="color:undefined;font-size:undefined;"></i></div>
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
      <div placeholder="请输入密码" maxlength="6" autocomplete="off" class="h-cell h-cell--required h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>密码</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="password" placeholder="请输入密码" maxlength="6" autocomplete="off" value="" class="h-field__control">
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="请输入姓名" class="h-cell h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>姓名</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="input" placeholder="请输入姓名" value="" class="h-field__control">
            <!---->
            <!---->
            <!---->
          </div>
          <div class="h-hairline--bottom h-field__error-message">您填写的姓名有误</div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="输入框已经被禁用" disabled="disabled" left-icon="contact" class="h-cell h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>姓名</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="input" placeholder="输入框已经被禁用" disabled="disabled" left-icon="contact" value="" class="h-field__control">
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="请输入姓名" class="h-cell h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>姓名</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="input" placeholder="请输入姓名" value="" class="h-field__control">
            <!---->
            <div class="h-field__icon"><i class="h-icon h-icon-question" style="color:undefined;font-size:undefined;"></i></div>
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
      <div placeholder="请输入" rows="1" class="h-cell h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>textarea</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><textarea placeholder="请输入" rows="1" class="h-field__control"></textarea>
            <!---->
            <!---->
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div>
    <div class="h-cell-group h-hairline--top-bottom">
      <div placeholder="请输入姓名" class="h-cell h-cell--center h-hairline h-field">
        <!---->
        <div class="h-cell__title"><span>姓名</span>
          <!---->
        </div>
        <div class="h-cell__value">
          <div class="h-field__body"><input type="input" placeholder="请输入姓名" value="" class="h-field__control">
            <!---->
            <!---->
            <div class="h-field__button"><button class="h-button h-button--primary h-button--small">
                <!----> <span class="h-button__text">重新发送</span></button></div>
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
  </div>
</div>
`;
